airflow/operators/docker_operator.py

```diff
@@ -124,21 +136,27 @@ def __init__(
         self.volumes = volumes or []
         self.xcom_push = xcom_push
         self.xcom_all = xcom_all
+        self.dockercfg_path = dockercfg_path
+        self.registry_username = registry_username
+        self.clear_volumes = clear_volumes
 
         self.cli = None
         self.container = None
 
+        if cpus is not None:
+            logging.warning("Ignoring the cpus parameter. Use cpu_shares instead.")
```

Comment:
The issue is that we must make sure we maintain backwards-compatibility for Airflow users. The best practice is to mark cpu's a deprecated but still allow it and give a warning that in the near future it will no longer be allowed. Dropping it outright will break existing DAGs.